### ISSUE TITLE
Tun error

### ISSUE THREAD

**Issue by drdre-lab** [engageub/InternetIncome#173]
Please help me. I'm trying to launch Honeygain using Test script with proxy. But the container with Tun2socks cannot start. Errors in the screenshot.
![Скриншот 30 11 24_16 32 52](https://github.com/user-attachments/assets/99e05cd8-79ba-4b80-8fd1-1a5f88fd953a)
I'm trying it on a freshly installed Ubuntu. I tried using the root user and the sudo user. I tried it on another clean VPS. I tried Tun2proxy instead of Tu2socks. A similar error remains.
![Скриншот 30 11 24_16 52 50](https://github.com/user-attachments/assets/72a0d69f-2946-42dd-8ef4-6bbbc16983c9)
I tried using http proxy instead of socks. I tried disabling dns in the settings. I think I've already tried everything.

**Comment by engageub** [comment]
Hello,
It looks like /dev/net/tun does not have required permissions on your VPS.
Please see https://github.com/xjasonlyu/tun2socks/issues/125 for more details.
You may change the permissions as mentioned and try to start. 

Thank you

**Comment by drdre-lab** [comment]
I leave a comment on my problem. I suffered for a couple of days, but decided. The problem was in the newest version of the Docker. I installed clean Ubuntu 22 and installed the latest Docker from Docker repositories (29). But apparently in the latest version there is a conflict with Ubuntu 22 - containers could not launch tun. Although I performed and run all the actions on behalf of root. Then I installed Docker with `sudo apt-get -y install docker.io`. The old version from Ubuntu repositories has been installed (24). And this is the working system. 
![01 12 24_19 08 01](https://github.com/user-attachments/assets/91872b80-539f-441c-95c6-296bf8508daf)
The problem is gone, everything works like clockwork. Thank you for the script.

**Comment by engageub** [comment]
Thank you for the update. So you have installed docker separately and not by using the command mentioned which caused this issue?
The steps mentioned in the Readme are for the new Ubuntu VPS after getting it from the provider. 
It updates the OS and then installs docker. 
If you get a new VPS, just follow the commands mentioned in ReadMe file. 

Thank you

**Comment by drdre-lab** [comment]
> So you have installed docker separately and not by using the command mentioned which caused this issue?
Yes, initially I installed Docker of newest version separately on the new VPS. And that was my fatal mistake. The installed newest Docker and Ubuntu 22 did not allow Tun2socks to start.

**Comment by theo-bnts** [comment]
Thanks to the owner for this work!
I had the same problem with an old Docker installation but I couldn't reinstall Docker.
Here's the solution I found: Replace `-v '/dev/net/tun:/dev/net/tun'` with `--device /dev/net/tun`.

**Comment by engageub** [comment]
@theo-bnts ,
Thank you for the update. Other users in the issues mentioned were having the same problem.
Some users had to use `--privileged` flag along with the mentioned command to use device.
However if you wish to install docker. You can simply use the built-in command in the script.
`sudo bash internetIncome --install`  . This command worked for another user in the recent issue https://github.com/engageub/InternetIncome/issues/177    

Similar Issues from other docker users below:
https://github.com/haugene/docker-transmission-openvpn/issues/2900
https://github.com/qdm12/gluetun/issues/2606
https://github.com/haugene/docker-transmission-openvpn/issues/2883
